vmhost/contexts/validator.go

```diff
@@ -62,6 +62,20 @@ func (validator *wasmValidator) verifyProtectedFunctions(instance executor.Insta
 }
 
 func (validator *wasmValidator) verifyValidFunctionName(functionName string) error {
+	err := verifyCallFunction(functionName)
+	if err != nil {
+		return err
+	}
+
+	errInvalidName := fmt.Errorf("%w: %s", vmhost.ErrInvalidFunctionName, functionName)
```

Comment:
maybe `%x` for the function name? As it covers the situation where there are not ASCII characters involved